hq/test/logic/IamRemediationTest.scala

```diff
@@ -144,7 +145,39 @@ class IamRemediationTest extends FreeSpec with Matchers {
   }
 
   "lookupCredentialId" - {
-    "TODO" ignore {}
+    val nonMatchingAccessKey = CredentialMetadata("adam.fisher", "AKIAIOSFODNN1EXAMPLE", date.minusDays(1), CredentialActive)
+    val matchingAccessKey = CredentialMetadata("amina.adewusi", "AKIAIOSFODNN2EXAMPLE", date, CredentialActive)
+    val matchingAccessKey2 = CredentialMetadata("amina.adewusi", "AKIAIOSFODNN3EXAMPLE", date, CredentialActive)
+
+    "given a key creation date matches a date in the metadata, return the correct metadata" in {
+      val result = lookupCredentialId(date, List(matchingAccessKey, nonMatchingAccessKey))
+      result.value.username shouldEqual matchingAccessKey.username
+    }
+    "given there are no matching key creation dates, return a failure" in {
+      val result = lookupCredentialId(date, List(nonMatchingAccessKey))
+      result.isFailedAttempt() shouldBe true
+    }
+    // both user's access keys sharing the exact same date is an edge case, because the creation date is accurate to the second
+    // and it's unlikely both keys would be created at exactly the same time, but could happen, especially if created using the CLI.
+    // If this happens then we won't know how to identify the keys' id, which is required to disable it, so we return a Failure.
+    "given a key creation date matches two dates in the metadata, return a failure" in {
+      val result = lookupCredentialId(date, List(matchingAccessKey2, matchingAccessKey))
+      result.isFailedAttempt() shouldBe true 
+    }
+    // AWS's credential's report shows that a key's last rotated date is recorded to the second,
+    // so this function must seek to match to the dates up to the second, but not the millisecond.
+    "given a key creation date matches to the minute, but not the second, return a failure" in {
+      val keyCreationDate = new DateTime(1,1,1,1,1,1)
+      val metaDataCreationDate = new DateTime(1,1,1,1,1,2)
+      val result = lookupCredentialId(keyCreationDate, List(matchingAccessKey.copy(creationDate = metaDataCreationDate)))
+      result.isFailedAttempt() shouldBe true
+    }
+    "given a key creation date matches to the second, but not the millisecond, return the metadata" in {
```

Comment:
I think I'd consider saying a little more here in the test message? We know the input data only has second resolution, so we're testing an edge case where the milliseconds unexpectedly don't match. I don't feel strongly here though, see what you think!